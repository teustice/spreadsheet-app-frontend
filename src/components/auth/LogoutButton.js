import React, {Component} from 'react';
import { connect } from 'react-redux';

import { setCurrentUser } from '../../actions/setCurrentUser'

class LogoutButton extends Component {

    resetCurrentUser() {
      localStorage.removeItem('currentUser')
      this.props.setCurrentUser(null);
      this.props.history.push('/signin')
    }


    render() {
        return (
          <React.Fragment>
            {this.props.currentUser &&
              <div className="logout-button">
                <button className="btn btn-sm" onClick={this.resetCurrentUser.bind(this)}>Log Out</button>
              </div>
            }
          </React.Fragment>
        );
    }
}


const mapStateToProps = state => ({
 ...state
})

const mapDispatchToProps = dispatch => ({
  setCurrentUser: (val) => dispatch(setCurrentUser(val)),
})

export default connect(mapStateToProps, mapDispatchToProps)(LogoutButton);
