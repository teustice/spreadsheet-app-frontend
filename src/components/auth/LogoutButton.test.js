import React from "react";
import {configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import {LogoutButton} from "./LogoutButton";

configure({ adapter: new Adapter() });

describe('<LogoutButton />', () => {

  describe('render()', () => {

    test('renders the component', () => {
      expect(toJson(<LogoutButton />)).toMatchSnapshot();
    });

    test('renders button when a user is logged in', () => {
      const wrapper = shallow(<LogoutButton currentUser={{email: 'test@test.com'}} />);
      expect(wrapper.find('button').text()).toEqual(`Log Out`);
    });


    test('It displays nothing without a logged in user', () => {
      const wrapper = shallow(<LogoutButton />);
      expect(wrapper.find('p').exists()).toEqual(false);
    });
  });

  describe('resetCurrentUser()', () => {

    test('It sets currentUser to null', () => {
      let user = 'some user';
      let history = [];
      let testSetCurrentUser = (value) => {
        user = value;
      }
      const wrapper = shallow(<LogoutButton history={history} setCurrentUser={testSetCurrentUser}/>);
      const instance = wrapper.instance();
      instance.resetCurrentUser()
      expect(user).toEqual(null);
    });


    test('It redirects to /signin', () => {
      let user = 'some user';
      let history = [];
      let testSetCurrentUser = (value) => {
        user = value;
      }
      const wrapper = shallow(<LogoutButton history={history} setCurrentUser={testSetCurrentUser}/>);
      const instance = wrapper.instance();
      instance.resetCurrentUser()
      expect(history[0]).toEqual('/signin');
    });

  });

});
