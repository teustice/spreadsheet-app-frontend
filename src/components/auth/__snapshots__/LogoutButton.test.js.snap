// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutButton /> render() renders the component 1`] = `undefined`;
